Data/IntMap/Internal.hs

```diff
@@ -3396,3 +3402,50 @@ node = "+--"
 withBar, withEmpty :: [String] -> [String]
 withBar bars   = "|  ":bars
 withEmpty bars = "   ":bars
+
```

Comment:
Should these assertions be here or under `tests`? I'm asking because I don't know.

Reply:
I was following structure that Data.Set used[1], but I'm open to moving them somewhere else; perhaps in a follow-up PR though.

[1] https://github.com/haskell/containers/blob/master/Data/Set/Internal.hs#L1758

Reply:
Looking through the implementations I don't think there's anything that requires them to be defined here since the internals of the types are exposed by the `Internal` module. I think I would prefer to do a mass move across all the data structures in a follow-up though.